Docs: Update template variable overview to contain the preferred variable syntax (#25176)
grafana/grafana: docs/sources/variables/templates-and-variables.md

- `$varname`
  This syntax is easy to read, but it does not allow you to use a variable in the middle of a word.
  **Example:** apps.frontend.$server.requests.count

> - `${var_name}` This syntax is needed when you want to interpolate a variable in the middle of an expression. 

Reviewer: ```suggestion
- `${var_name}` Use this syntax when you want to interpolate a variable in the middle of an expression. 
```

> - `${var_name:<format>}` Gives you more control over how values are interpolated. Read [Advanced variable format options]({{< relref "advanced-variable-format-options.md" >}}) for more detail on all the formatting types.

Reviewer: ```suggestion
- `${var_name:<format>}` This format gives you more control over how Grafana interpolates values. Refer to [Advanced variable format options]({{< relref "advanced-variable-format-options.md" >}}) for more detail on all the formatting types.
```

> - `[[varname]]` Deprecated old syntax, will be removed in a future release. 

Reviewer: ```suggestion
- `[[varname]]` Do not use. Deprecated old syntax, will be removed in a future release. 
```

Reviewer: I'm told this syntax is necessary for certain use cases. @daniellee , can you confirm?

Author: no it's not, all those use cases is covered by  ${variable}

Reviewer 2: The issue is that there are thousands of dashboards using this the [[ ]] syntax so I don't think we will ever remove it. Or what do you say @torkelo ?

Author: probably not :) maybe if we have a migration for it, but that is a bit tricky
